.changeset/five-dolphins-taste.md

```diff
@@ -0,0 +1,5 @@
+---
+'@compiled/eslint-plugin': minor
+---
+
+add engines to formalize supported node versions - ^16.0.0 || >= 18.0.0
```

Comment:
nit: the two changesets can be combined into one

Reply:
Oh, ok.  I was trying to keep the commits tidy (1st change -> 1st changeset -> 2nd change -> 2nd changeset).  Happy to re-organize it however you prefer.

Reply:
nah no need to re-organise unless you need to make other changes

probs https://github.com/atlassian-labs/compiled/pull/1760#discussion_r1889504901 is the only thing i'm not 100% confident with, otherwise PR looks good to me :)